src/main/java/baritone/utils/schematic/format/defaults/LitematicaSchematic.java

```diff
@@ -0,0 +1,296 @@
+     * @return the lower cord of the requested axis.
+     */
+    private static int getMinimumCord(NBTTagCompound nbt, String subReg, String s) {
+        int a = nbt.getCompoundTag(reg).getCompoundTag(subReg).getCompoundTag(pos).getInteger(s);
+        int b = nbt.getCompoundTag(reg).getCompoundTag(subReg).getCompoundTag(size).getInteger(s);
+        if (b < 0) {
+            b++;
+        }
+        return Math.min(a,a+b);
+
+    }
+
+    /**
+     * @param blockStatePalette List of all different block types used in the schematic.
+     * @return Array of BlockStates.
+     */
+    private static IBlockState[] getBlockList(NBTTagList blockStatePalette) {
+        IBlockState[] blockList = new IBlockState[blockStatePalette.tagCount()];
+
+        for (int i = 0; i< blockStatePalette.tagCount(); i++) {
+            Block block = Block.REGISTRY.getObject(new ResourceLocation((((NBTTagCompound) blockStatePalette.get(i)).getString("Name"))));
+            NBTTagCompound properties = ((NBTTagCompound) blockStatePalette.get(i)).getCompoundTag("Properties");
+
+            blockList[i] = getBlockState(block, properties);
+        }
+        return blockList;
+    }
+
+    /**
+     * @param block block.
+     * @param properties List of Properties the block has.
+     * @return A blockState.
+     */
+    private static IBlockState getBlockState(Block block, NBTTagCompound properties) {
+        IBlockState blockState = block.getDefaultState();
+
+        for (Object key : properties.getKeySet().toArray()) {
+            IProperty<?> property = block.getBlockState().getProperty((String) key);
+            String propertyValue = properties.getString((String) key);
+            if (property != null) {
+                blockState = setPropertyValue(blockState, property, propertyValue);
+            }
+        }
+        return blockState;
+    }
+
+    /**
+     * i haven't written this and i wont try to decode it.
+     * @param state .
+     * @param property .
+     * @param value .
+     * @return .
+     * @param <T> .
+     */
+    private static <T extends Comparable<T>> IBlockState setPropertyValue(IBlockState state, IProperty<T> property, String value) {
+        Optional<T> parsed = property.parseValue(value).toJavaUtil();
+        if (parsed.isPresent()) {
+            return state.withProperty(property, parsed.get());
+        } else {
+            throw new IllegalArgumentException("Invalid value for property " + property);
+        }
+    }
+
+    /**
+     * @param amountOfBlockTypes amount of block types in the schematic.
+     * @return amount of bits used to encode a block.
+     */
+    private static int getBitsPerBlock(int amountOfBlockTypes) {
+        return  (int) Math.floor((Math.log(amountOfBlockTypes)) / Math.log(2))+1;
+    }
+
+    /**
+     * @return the volume of the subregion.
+     */
+    private static long getVolume(NBTTagCompound nbt, String subReg) {
+        return Math.abs(
```

Comment:
add comment explaining why abs is needed (because size is sometimes negative?)

Reply:
exactly if a  uneven amount of axis are negative the volume would be negative, but there can only be positive volumes